persist/fs/commitlog/writer.go

```diff
@@ -154,19 +157,38 @@ func (w *writer) isOpen() bool {
 	return w.chunkWriter.fd != nil
 }
 
+func (w *writer) logEntryIndex(seriesUniqueIndex uint64) uint64 {
+	// Special case the zeroth value as we compare the value
+	// at the entry of the seen cache with the index being put
+	// at the location - since all entries in the seen cache when
+	// not set are zero the value being used to compare whether
+	// this result has been seen or not cannot be zero itself
+	return seriesUniqueIndex + 1
+}
+
 func (w *writer) Write(
 	series Series,
 	datapoint ts.Datapoint,
 	unit xtime.Unit,
 	annotation ts.Annotation,
 ) error {
+	logEntryIndex := w.logEntryIndex(series.UniqueIndex)
+
 	var logEntry schema.LogEntry
 	logEntry.Create = w.nowFn().UnixNano()
-	logEntry.Index = series.UniqueIndex
-
-	seen := w.bitset.has(logEntry.Index)
-	if !seen {
-		// If "idx" hasn't been written to commit log
+	logEntry.Index = logEntryIndex
+
+	// NB(r): Calculate whether metadata for this series
+	// has been written before. Once we have more unique series than
+	// the size of the seen cache this can cause some metadata
+	// to be written multiple times. The vast majority
+	// of IDs should not collide frequently which preserves
+	// the size of the resulting commit log staying small.
+	seenIdx := logEntryIndex % uint64(w.seenCacheLen)
```

Comment:
Just throwing it out there, would a bloom filter help reduce the false positive rate here?